lldb/packages/Python/lldbsuite/test/make/Makefile.rules

```diff
@@ -456,21 +492,15 @@ ifeq (1, $(USE_SYSTEM_STDLIB))
         endif
         CXXFLAGS += -nostdlib++ -nostdinc++ -cxx-isystem $(SDKROOT)/usr/include/c++/v1
         LDFLAGS += -L$(SDKROOT)/usr/lib -Wl,-rpath,$(SDKROOT)/usr/lib -lc++
+    else
+        ifneq (,$(findstring clang,$(CC)))
+            # Force clang looking for the gcc's headers at specific rootfs folder.
+            CXXFLAGS += -stdlib=libstdc++ $(GCC_TOOLCHAIN_FLAGS)
```

Comment:
..and then change this to not pass any flags by default? (I don't know what `-stdlib=libstdc++` will do for windows, but I don't expect it to go down well)

Reply:
This still forces the usage of libstdc++ on windows, right? Could we just pass nothing here and let clang use its default?

Reply:
Agreed

Reply:
> This still forces the usage of libstdc++ on windows, right? Could we just pass nothing here and let clang use its default?

Actually, it's ignored on Windows. We have some problems without that line during cross-compiling, but I think it's due to missing $(GCC_TOOLCHAIN_FLAGS) content, not stdlib selection.

Reply:
> This still forces the usage of libstdc++ on windows, right? Could we just pass nothing here and let clang use its default?

Sorry, that was too early. 
We still need this flag. It is ignored for Windows target, but clang tries to pass -lc++ to linker in case of Linux target if -lstdc++ is not specified, and ld.lld fails to find it (since it's not present in my sysroot).

Reply:
Okay, but that is a clang configuration thing (`CLANG_DEFAULT_CXX_STDLIB` cmake option), right? Can't you set it to the actual c++ library that you intend to use instead of overriding it here? Or use a clang wrapper or a [configuration file](https://clang.llvm.org/docs/UsersManual.html#configuration-files) to override it externally? (FWIW, that's what we do internally to adapt the test suite to our (strange) internal infrastructure)

It's nice (maybe) that `-lstdc++` is ignored on windows, but I still this this is the wrong thing to do. `libstdc++` is definitely not the system default everywhere.

Reply:
Removed `-stdlib=libstdc++`. The problem was, that we were trying to link with in-tree libcxx not linked statically to libcxxabi, but clang driver couldn't use it without the explicit -lc++abi flag (it doesn't distinguish libcxx comilation types according to https://releases.llvm.org/6.0.0/projects/libcxx/docs/BuildingLibcxx.html#using-libcxxrt-on-linux).
Adding LIBCXX_ENABLE_STATIC_ABI_LIBRARY helps.

Reply:
That makes sense. The fix sounds good to me, but what does that mean for #99589 ? Is that compatible with this? Should we drop it?

Reply:
Seemingly we don't need it anymore. I launched the testing on our CI. As soon as it shows that it makes no difference without -lc++abi, I'll close #99589.